roachtest: npgsql failed [crashes with unsupported binary serialization of multidimensional arrays]

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13656934?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13656934?buildTab=artifacts#/npgsql) on master @ [585ffa0acfa6dd771aad8320d3a9e219acf56656](https://github.com/cockroachdb/cockroach/commits/585ffa0acfa6dd771aad8320d3a9e219acf56656):


```
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(Multiplexing).Send_unknown -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1E+24::numeric",1000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(NonMultiplexing,Jsonb).Write_as_ArraySegment_of_char -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V2,DefaultReplicationDataMode,StreamingTransactionMode).Truncate(Cascade) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("1111.2222333344445555666677778::numeric",1111.2222333344445555666677778m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("1E+28::numeric",10000000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Read_type_with_more_properties_than_attributes -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("-2.5::numeric(10,0)",-3m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("-1.4::numeric(10,0)",-1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("-2.4::numeric(10,0)",-2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTypeTests(NonMultiplexing).Double -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,DefaultTransactionMode).CompositeType -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("100000000000000000000::numeric",100000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1000000000000000000000000::numeric",1000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1E-4::numeric",0.0001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_type_without_getter_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Types.TextTests(NonMultiplexing).Text_as_array_of_chars -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("-2.5::numeric(10,0)",-3m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1111.2222333344445555666677778::numeric",1111.2222333344445555666677778m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("100000000::numeric",100000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Read_type_with_private_setter_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("+79228162514264337593543950335::numeric",decimal.MaxValue) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Date_as_DateOnly -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1.5::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("0.000000000001::numeric",0.000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("1.5::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.FunctionTests.Void -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1.2::numeric(10,4)",1.2000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_type_without_setter -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("1000000000000000000000000::numeric",1000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("-2.4::numeric(10,0)",-2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(NonMultiplexing).Write_Record_is_not_supported -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("0.0001::numeric",0.0001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("0.0001::numeric",0.0001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("10000000000000000::numeric",10000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Datemultirange_as_array_of_NpgsqlRange_of_DateTime -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ArrayTests(NonMultiplexing).IEnumerable_throws_friendly_exception -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("2.4::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("-1.5::numeric(10,0)",-2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("0.000000000001::numeric",0.000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_struct_with_property -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("100000000000000000000::numeric",100000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(Multiplexing).Boolean -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V1,BinaryReplicationDataMode,DefaultTransactionMode).TextReader -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Timestamp_as_long -  (unexpected)
--- FAIL: Npgsql.Tests.FunctionTests.DeriveParameters_non_existing_function -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,DefaultTransactionMode).LogicalDecodingMessage(False) -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13656934/npgsql/1706020185997/1706020475108)

</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35609

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13669769?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13669769?buildTab=artifacts#/npgsql) on master @ [2bc9c820cd35cc4756a408a51efab005d4b48604](https://github.com/cockroachdb/cockroach/commits/2bc9c820cd35cc4756a408a51efab005d4b48604):


```
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V1,BinaryReplicationDataMode,DefaultTransactionMode).TwoPhase(False) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_type_with_more_properties_than_attributes -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("1E+20::numeric",100000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("2::numeric(10,2)",2.00m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("-2.4::numeric(10,0)",-2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("-1.4::numeric(10,0)",-1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NetworkTypeTests(NonMultiplexing).Inet_v6_as_IPAddress -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("2.4::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("10000::numeric",10000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("1::numeric(10,2)",1.00m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("+79228162514264337593543950335::numeric",decimal.MaxValue) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("0.0000000000000001::numeric",0.0000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NetworkTypeTests(Multiplexing).IPAddress_Any -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("111.22223333444455556666777788::numeric",111.22223333444455556666777788m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(NonMultiplexing).BitVector32 -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("2::numeric(10,2)",2.00m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Read_type_with_private_setter_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Types.LegacyDateTimeTests.Timestamptz_local_DateTime_converts -  (unexpected)
--- FAIL: Npgsql.Tests.Types.TextTests(Multiplexing).Text_as_ArraySegment_of_chars -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("1E+16::numeric",10000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.CommonReplicationTests<LogicalReplicationConnection>.Synchronous_replication -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("1.2::numeric(10,2)",1.20m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("10000000000000000000000000000::numeric",10000000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1000000000000::numeric",1000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_overflow_is_safe -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,StreamingTransactionMode).Truncate(None) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("0.0036882500000000000000000000",0.0036882500000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("-2.4::numeric(10,0)",-2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("-1.5::numeric(10,0)",-2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NetworkTypeTests(Multiplexing).Inet_v6_as_NpgsqlInet -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_type_with_two_properties_inverted -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeTests(NonMultiplexing).Table_as_composite(False) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("0::numeric",0m) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V2,DefaultReplicationDataMode,StreamingTransactionMode).Stream -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("1E+20::numeric",100000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ArrayTests(Multiplexing).IList_non_generic -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V1,BinaryReplicationDataMode,DefaultTransactionMode).Update_for_index_replica_identity -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Timestamptz_as_long -  (unexpected)
--- FAIL: Npgsql.Tests.Types.TextTests(NonMultiplexing).Text_as_array_of_bytes -  (unexpected)
--- FAIL: Npgsql.Tests.PoolTests.Timeout_getting_connector_from_exhausted_pool(True) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1.5::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("1234567844445555.000000000",1234567844445555.000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("1E-12::numeric",0.000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("0.0001::numeric",0.0001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(Multiplexing).DbType_causes_inference -  (unexpected)
--- FAIL: Npgsql.Tests.SecurityTests.DataSource_UserCertificateValidationCallback_is_invoked(False) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("10000000000000000::numeric",10000000000000000m) -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_fs=ext4</code>
 - <code>ROACHTEST_localSSD=true</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13669769/npgsql/1706045665166/1706045945849)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13672759?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13672759?buildTab=artifacts#/npgsql) on master @ [e0fbbe4c1334580bf05d6d6fb0d406a327e43944](https://github.com/cockroachdb/cockroach/commits/e0fbbe4c1334580bf05d6d6fb0d406a327e43944):


```
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Timestamp_as_long -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,DefaultTransactionMode).Delete_for_default_replica_identity -  (unexpected)
--- FAIL: Npgsql.Tests.Types.FullTextSearchTests(NonMultiplexing).TsQuery("'a' <-> 'b'",'a' <-> 'b') -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1E+12::numeric",1000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(Multiplexing).Bitstring_with_multiple_bits_as_bool_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("936490726837837729197",936490726837837729197m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NetworkTypeTests(Multiplexing).Inet_v6_as_tuple -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,StreamingTransactionMode).Truncate(None) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("+79228162514264337593543950335::numeric",decimal.MaxValue) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_class_with_field -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NetworkTypeTests(NonMultiplexing).IPAddress_Any -  (unexpected)
--- FAIL: Npgsql.Tests.AuthenticationTests(NonMultiplexing).Use_pgpass_from_homedir -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("1E+20::numeric",100000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("0.00000000000000000001::numeric",0.00000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("1E-20::numeric",0.00000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(NonMultiplexing).Bitstring_with_multiple_bits_as_bool_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Date_as_int -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(NonMultiplexing).Array_of_single_bits -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("1::numeric(10,2)",1.00m) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V2,DefaultReplicationDataMode,StreamingTransactionMode).TwoPhase(False) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("2.4::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_type_with_less_properties_than_attributes_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("9364907268378377291970000",9364907268378377291970000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Read_class_with_property -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("1E+16::numeric",10000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,DefaultTransactionMode).LogicalDecodingMessage(False) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(NonMultiplexing).AllResultTypesAreUnknown -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1E-20::numeric",0.00000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("-2.4::numeric(10,0)",-2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("2.4::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("-2.5::numeric(10,0)",-3m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("2.4::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.AuthenticationTests(NonMultiplexing).Use_pgpass_from_connection_string -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,DefaultTransactionMode).TextReader -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,StreamingTransactionMode).Insert -  (unexpected)
--- FAIL: Npgsql.Tests.StoredProcedureTests.DeriveParameters_procedure_various -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ArrayTests(NonMultiplexing).Jagged_arrays_not_supported -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("-79228162514264337593543950335::numeric",decimal.MinValue) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ArrayTests(Multiplexing).Bind_int_then_array_of_int -  (unexpected)
--- FAIL: Npgsql.Tests.Types.TextTests(NonMultiplexing).Text_as_string -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTypeTests(NonMultiplexing).Read_overflow<Int64>(0L,9.223372036854776E+18d,"decimal") -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("1111.2222333344445555666677778::numeric",1111.2222333344445555666677778m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("100000000::numeric",100000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("11112222000000000000",11112222000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ByteaTests(Multiplexing).Write_as_NonSeekable_stream -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NetworkTypeTests(Multiplexing).IPAddress_Any -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("1::numeric(10,2)",1.00m) -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13672759/npgsql/1706107069455/1706107348360)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13685746?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13685746?buildTab=artifacts#/npgsql) on master @ [e996b2f9587ec45bc2e419603ea95b86b4d1e379](https://github.com/cockroachdb/cockroach/commits/e996b2f9587ec45bc2e419603ea95b86b4d1e379):


```
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,StreamingTransactionMode).Update_for_full_replica_identity -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("0.00000001::numeric",0.00000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.SecurityTests.DataSource_UserCertificateValidationCallback_is_invoked(False) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("1000000000000::numeric",1000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(Multiplexing,Jsonb).As_string -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ByteaTests(NonMultiplexing).Write_as_MemoryStream_truncated -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_type_without_getter_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V1,BinaryReplicationDataMode,DefaultTransactionMode).Update_for_index_replica_identity -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("1::numeric",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1E-28::numeric",0.0000000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("100000000000000000000::numeric",100000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1E+20::numeric",100000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("2.4::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Interval_as_NpgsqlInterval -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,DefaultTransactionMode).Truncate(RestartIdentity) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1E+8::numeric",100000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("100000000000000000000::numeric",100000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1.2::numeric(10,4)",1.2000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V1,BinaryReplicationDataMode,DefaultTransactionMode).CompositeType -  (unexpected)
--- FAIL: Npgsql.Tests.Types.FullTextSearchTests(Multiplexing).TsQuery("'a'",'a') -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(Multiplexing,Jsonb).As_poco_long -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1.2::numeric(10,1)",1.2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ByteaTests(Multiplexing).Write_as_ArraySegment -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(NonMultiplexing).Write_as_string -  (unexpected)
--- FAIL: Npgsql.Tests.CommandTests(NonMultiplexing).Named_parameters_are_not_supported_when_EnableSqlParsing_is_disabled -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("1::numeric",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_type_with_private_getter_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Cannot_mix_DateTime_Kinds_in_array -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("9364907268378377291970000",9364907268378377291970000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,StreamingTransactionMode).ValueMetadata -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,StreamingTransactionMode).Truncate(Cascade, RestartIdentity) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("1000000000000::numeric",1000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("10000000000000000000000000000::numeric",10000000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.MultipleHostsTests.IntegrationTest(True,True) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("-1.5::numeric(10,0)",-2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(NonMultiplexing,Jsonb).Write_as_ArraySegment_of_char -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1E-28::numeric",0.0000000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_struct_with_property -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(Multiplexing).Read_unknown -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).BigInteger_large -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1E+4::numeric",10000m) -  (unexpected)
--- FAIL: Npgsql.Tests.SecurityTests.SslMode_Require_throws_without_TSC -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(NonMultiplexing,Jsonb).As_string_with_GetTextReader -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ArrayTests(NonMultiplexing).Read_two_empty_arrays -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("+79228162514264337593543950335::numeric",decimal.MaxValue) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("1.2::numeric(10,1)",1.2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("0.0001::numeric",0.0001m) -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13685746/npgsql/1706194232376/1706194518120)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13700342?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13700342?buildTab=artifacts#/npgsql) on master @ [45cd83b6838bd515c273955c90ade06ca9fa8794](https://github.com/cockroachdb/cockroach/commits/45cd83b6838bd515c273955c90ade06ca9fa8794):


```
--- FAIL: Npgsql.Tests.Types.TextTests(Multiplexing).Aliased_DbTypes(AnsiStringFixedLength) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.TextTests(NonMultiplexing).Truncate -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(NonMultiplexing).UnknownResultTypeList -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("3649072683783772919700000000",3649072683783772919700000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("1E-24::numeric",0.000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(NonMultiplexing).Uuid -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("10000::numeric",10000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,DefaultTransactionMode).TwoPhase(False) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1.2::numeric(10,5)",1.20000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1E+0::numeric",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V1,BinaryReplicationDataMode,DefaultTransactionMode).Update_for_default_replica_identity -  (unexpected)
--- FAIL: Npgsql.Tests.ConnectionTests(NonMultiplexing).Breaking_connection_while_loading_database_info -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1000000000000::numeric",1000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.TextTests(NonMultiplexing).Text_as_array_of_chars -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(NonMultiplexing).Write_as_string_validation -  (unexpected)
--- FAIL: Npgsql.Tests.SecurityTests.Connection_UserCertificateValidationCallback_is_invoked(False) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("1::numeric(10,2)",1.00m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("1E+12::numeric",1000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("1::numeric",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(NonMultiplexing).Unsupported_DbType -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("-79228162514264337593543950335::numeric",decimal.MinValue) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V2,DefaultReplicationDataMode,StreamingTransactionMode).CompositeType -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("1E-16::numeric",0.0000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("1.2::numeric(10,3)",1.200m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("1E-20::numeric",0.00000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("0.0001::numeric",0.0001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1000000000000::numeric",1000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("1E+28::numeric",10000000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("2.4::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.ConnectionTests(NonMultiplexing).Client_encoding_is_UTF8_by_default -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(NonMultiplexing).BitVector32_too_long -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTypeTests(Multiplexing).Int64 -  (unexpected)
--- FAIL: Npgsql.Tests.PoolTests.Timeout_getting_connector_from_exhausted_pool(True) -  (unexpected)
--- FAIL: Npgsql.Tests.PoolTests.ClearPool(1) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("11112222000000000000",11112222000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(NonMultiplexing).DbType_causes_inference -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(Multiplexing).Write_as_string -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(Multiplexing,Jsonb).As_char_array -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("1E+12::numeric",1000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Interval_with_microsecond -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Read_type_with_two_parameters_reversed -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1E+24::numeric",1000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("0.0000000000000000000000000001::numeric",0.0000000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("2.4::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.ConnectionTests(Multiplexing).Timezone_env_var -  (unexpected)
--- FAIL: Npgsql.Tests.ConnectionTests(NonMultiplexing).PhysicalConnectionInitializer_async_with_break -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ArrayTests(NonMultiplexing).Read_null_as_non_nullable_array_throws -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13700342/npgsql/1706279838239/1706280122045)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13713791?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13713791?buildTab=artifacts#/npgsql) on master @ [b59a83a1cd31d8e00f39cf32b1cf99b41b96e28e](https://github.com/cockroachdb/cockroach/commits/b59a83a1cd31d8e00f39cf32b1cf99b41b96e28e):


```
--- FAIL: Npgsql.Tests.Types.JsonTests(Multiplexing,Jsonb).Write_as_ArraySegment_of_char -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(NonMultiplexing,Jsonb).As_JsonDocument -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Datemultirange_as_array_of_NpgsqlRange_of_DateTime -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(NonMultiplexing).Boolean -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("-1.5::numeric(10,0)",-2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1E-8::numeric",0.00000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("2::numeric(10,2)",2.00m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeInfinityTests(True).Timestamp_write -  (unexpected)
--- FAIL: Npgsql.Tests.Types.TextTests(NonMultiplexing).Aliased_postgres_types("name",Name) -  (unexpected)
--- FAIL: Npgsql.Tests.StoredProcedureTests.With_output_parameters(False,True) -  (unexpected)
--- FAIL: Npgsql.Tests.PoolTests.ClearPool(2) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.LegacyDateTimeTests.Timestamptz_local_DateTime_converts -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,DefaultTransactionMode).Update_for_full_replica_identity -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("1.2::numeric(10,3)",1.200m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("100000000::numeric",100000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1E-28::numeric",0.0000000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("10000::numeric",10000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Read_type_with_private_setter_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_type_with_explicit_property_name -  (unexpected)
--- FAIL: Npgsql.Tests.SecurityTests.Connection_UserCertificateValidationCallback_is_invoked(True) -  (unexpected)
--- FAIL: Npgsql.Tests.ConnectionTests(NonMultiplexing).DatabaseInfo_is_shared -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("1E-16::numeric",0.0000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1111.2222333344445555666677778::numeric",1111.2222333344445555666677778m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("11.222233334444555566667777888::numeric",11.222233334444555566667777888m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1E-28::numeric",0.0000000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("10000000000000000000000000000::numeric",10000000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(Multiplexing).AllResultTypesAreUnknown -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeInfinityTests(True).Date_DateTime_write -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(NonMultiplexing).AllResultTypesAreUnknown -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("0.000000000000000000000001::numeric",0.000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("2::numeric(10,2)",2.00m) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,StreamingTransactionMode).CreatePgOutputReplicationSlot -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ArrayTests(NonMultiplexing).Zero_dimensional -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("0.0000000000000000000000000001::numeric",0.0000000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,StreamingTransactionMode).Truncate(None) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(Multiplexing,Jsonb).As_poco -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("1E-28::numeric",0.0000000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.TextTests(Multiplexing).Truncate -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V1,BinaryReplicationDataMode,DefaultTransactionMode).Truncate(None) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.TextTests(Multiplexing).Text_as_array_of_bytes -  (unexpected)
--- FAIL: Npgsql.Tests.ConnectionTests(Multiplexing).Timezone_env_var -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("-2.4::numeric(10,0)",-2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1::numeric",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("+79228162514264337593543950335::numeric",decimal.MaxValue) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeInfinityTests(True).Date_DateOnly_read -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(NonMultiplexing).Unsupported_DbType -  (unexpected)
--- FAIL: Npgsql.Tests.Types.TextTests(Multiplexing).Text_long -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13713791/npgsql/1706355264576/1706355552714)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13723973?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13723973?buildTab=artifacts#/npgsql) on master @ [0baf22a03d5f55e2611701bc723e3e0b713ab051](https://github.com/cockroachdb/cockroach/commits/0baf22a03d5f55e2611701bc723e3e0b713ab051):


```
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("2::numeric(10,2)",2.00m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("1.4::numeric(10,0)",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("-2.5::numeric(10,0)",-3m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_type_with_explicit_property_name -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1E-24::numeric",0.000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("1E+28::numeric",10000000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("10000::numeric",10000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("100000000::numeric",100000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("1E+4::numeric",10000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("2.4::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.TimeTz_before_utc_zero -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("936490726837837729197",936490726837837729197m) -  (unexpected)
--- FAIL: Npgsql.Tests.ConnectionTests(Multiplexing).Clone -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("10000::numeric",10000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ArrayTests(NonMultiplexing).Mixed_element_types -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1234567844445555.000000000",1234567844445555.000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.PoolTests.ClearPool_with_busy -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("1E+24::numeric",1000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.InternalTypeTests(Multiplexing).Read_internal_char -  (unexpected)
--- FAIL: Npgsql.Tests.NpgsqlEventSourceTests.Command_start_stop -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Read_type_with_less_parameters_than_attributes_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Read_type_with_less_properties_than_attributes_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,DefaultTransactionMode).Update_for_full_replica_identity -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(Multiplexing).Bitstring_with_multiple_bits_as_bool_throws -  (unexpected)
--- FAIL: Npgsql.Tests.MultipleHostsTests.BuildMultiHost_with_single_host_is_supported -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("0.0000000000000001::numeric",0.0000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("0.00000000000000000001::numeric",0.00000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("1.4::numeric(10,0)",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("0.0000000000000001::numeric",0.0000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(NonMultiplexing).Bitstring_with_multiple_bits_as_bool_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("100000000000000000000::numeric",100000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V2,DefaultReplicationDataMode,StreamingTransactionMode).Truncate(Cascade) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(Multiplexing).DbType_causes_inference -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Read_type_with_two_properties_inverted -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Numeric -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Cannot_mix_DateTime_Kinds_in_array -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("0::numeric",0m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("1E+28::numeric",10000000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(NonMultiplexing).Write_as_string_validation -  (unexpected)
--- FAIL: Npgsql.Tests.Types.TextTests(NonMultiplexing).Aliased_DbTypes(AnsiStringFixedLength) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ByteaTests(NonMultiplexing).Bytea -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1111.2222333344445555666677778::numeric",1111.2222333344445555666677778m) -  (unexpected)
--- FAIL: Npgsql.Tests.StoredProcedureTests.DeriveParameters_non_existing_procedure -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1E-4::numeric",0.0001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(Multiplexing,Jsonb).Can_read_two_json_documents -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("1::numeric",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("0.0000000000000000000000000001::numeric",0.0000000000000000000000000001m) -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13723973/npgsql/1706441344463/1706441621292)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118392 roachtest: npgsql failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-23.2.0-rc release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13731862?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13731862?buildTab=artifacts#/npgsql) on master @ [0baf22a03d5f55e2611701bc723e3e0b713ab051](https://github.com/cockroachdb/cockroach/commits/0baf22a03d5f55e2611701bc723e3e0b713ab051):


```
--- FAIL: Npgsql.Tests.Types.BitStringTests(NonMultiplexing).Read_as_string_is_not_supported -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V1,BinaryReplicationDataMode,DefaultTransactionMode).LogicalDecodingMessage(False) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ByteaTests(NonMultiplexing).Write_as_MemoryStream_long -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1E+12::numeric",1000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V1,BinaryReplicationDataMode,DefaultTransactionMode).TextReader -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Timestamptz_as_DateTimeOffset_utc -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,DefaultTransactionMode).Dispose_while_replicating -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1::numeric",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("-79228162514264337593543950335::numeric",decimal.MinValue) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V1,BinaryReplicationDataMode,DefaultTransactionMode).Update_for_default_replica_identity -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1.4::numeric(10,0)",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("-2.5::numeric(10,0)",-3m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("-2.4::numeric(10,0)",-2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(Multiplexing).BitArray_empty -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ArrayTests(NonMultiplexing).Read_two_empty_arrays -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("-1.4::numeric(10,0)",-1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1.2::numeric(10,2)",1.20m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("0.0000000000000000000000000001::numeric",0.0000000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.PoolManagerTests.With_canonical_connection_string -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("1E-16::numeric",0.0000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("100000000000000000000::numeric",100000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(Multiplexing,Jsonb).As_string_long -  (unexpected)
--- FAIL: Npgsql.Tests.Types.LegacyDateTimeTests.Timestamptz_write_utc_DateTime_does_not_convert -  (unexpected)
--- FAIL: Npgsql.Tests.SecurityTests.Bug1718 -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("2.4::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("2.5::numeric(10,0)",3m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ArrayTests(NonMultiplexing).IList_generic -  (unexpected)
--- FAIL: Npgsql.Tests.CommandTests(NonMultiplexing).Positional_parameters_are_supported_when_EnableSqlParsing_is_disabled -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(Multiplexing).Null -  (unexpected)
--- FAIL: Npgsql.Tests.MultipleHostsTests.IntegrationTest(True,False) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_type_without_setter -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Interval_as_NpgsqlInterval -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("1::numeric",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("11112222000000000000",11112222000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("936490726837837729197",936490726837837729197m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeInfinityTests(False).Date_DateOnly_write -  (unexpected)
--- FAIL: Npgsql.Tests.PoolTests.ClearPool(2) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V2,DefaultReplicationDataMode,StreamingTransactionMode).Truncate(Cascade, RestartIdentity) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V1,BinaryReplicationDataMode,DefaultTransactionMode).Truncate(RestartIdentity) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(NonMultiplexing,Jsonb).Can_read_two_json_documents -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTypeTests(Multiplexing).Read_overflow<Int64>(0L,9.223372036854776E+18d,"decimal") -  (unexpected)
--- FAIL: Npgsql.Tests.SecurityTests.Reject_self_signed_certificate(VerifyCA) -  (unexpected)
--- FAIL: Npgsql.Tests.SecurityTests.SslMode_Require_throws_without_TSC -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("1234567844445555.000000000",1234567844445555.000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.PoolTests.Timeout_getting_connector_from_exhausted_pool(True) -  (unexpected)
--- FAIL: Npgsql.Tests.TransactionTests(NonMultiplexing).Bug3686 -  (unexpected)
--- FAIL: Npgsql.Tests.Types.InternalTypeTests(Multiplexing).NpgsqlLogSequenceNumber -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13731862/npgsql/1706528000421/1706528279073)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118392 roachtest: npgsql failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-23.2.0-rc release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13747378?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13747378?buildTab=artifacts#/npgsql) on master @ [1d2534b831775dd84edcf7372d38e35ed4903278](https://github.com/cockroachdb/cockroach/commits/1d2534b831775dd84edcf7372d38e35ed4903278):


```
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(Multiplexing).Uuid -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Timestamp_cannot_use_as_DateTimeOffset -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(NonMultiplexing).Send_unknown -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("0.00000001::numeric",0.00000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Datemultirange_as_array_of_NpgsqlRange_of_DateOnly -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(NonMultiplexing,Jsonb).Can_read_two_json_documents -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("-2.4::numeric(10,0)",-2m) -  (unexpected)
--- FAIL: Npgsql.Tests.AuthenticationTests(Multiplexing).Connect_UserNameFromEnvironment_Succeeds -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,StreamingTransactionMode).TwoPhase(True) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_type_without_getter_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("-1.4::numeric(10,0)",-1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1000000000000::numeric",1000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ByteaTests(Multiplexing).Write_as_FileStream -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,StreamingTransactionMode).Bug4633 -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("+79228162514264337593543950335::numeric",decimal.MaxValue) -  (unexpected)
--- PASS: Npgsql.Tests.ReaderNewSchemaTests(Sync).IsAutoIncrement - unknown (unexpected)
--- FAIL: Npgsql.Tests.Types.NetworkTypeTests(NonMultiplexing).Inet_v4_as_tuple -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(NonMultiplexing).Write_as_string_validation -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.TimeTz_as_DateTimeOffset -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(Multiplexing).DbType_causes_inference -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("+79228162514264337593543950335::numeric",decimal.MaxValue) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Read_type_with_more_properties_than_attributes -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V1,BinaryReplicationDataMode,DefaultTransactionMode).Update_for_full_replica_identity -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NetworkTypeTests(Multiplexing).Inet_v4_as_NpgsqlInet -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_struct_with_property -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,DefaultTransactionMode).Update_for_default_replica_identity -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("0::numeric",0m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NetworkTypeTests(Multiplexing).Inet_v6_as_IPAddress -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Read_type_without_getter -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("1.2::numeric(10,3)",1.200m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(Multiplexing,Jsonb).As_char_array -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeInfinityTests(False).Date_DateTime_read -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ArrayTests(NonMultiplexing).Bind_int_then_array_of_int -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("1E+0::numeric",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("11112222000000000000",11112222000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("1234567844445555.000000000",1234567844445555.000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_type_with_more_properties_than_attributes -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("1::numeric",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read_BigInteger("-2.5::numeric(10,0)",-3m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("0.00000000000000000001::numeric",0.00000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ArrayTests(NonMultiplexing).Mixed_element_types -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("1E+0::numeric",1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.CompositeHandlerTests.Write_type_with_less_properties_than_attributes_throws -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("1E+12::numeric",1000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(Multiplexing,Jsonb).As_poco_long -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("11.222233334444555566667777888::numeric",11.222233334444555566667777888m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("-1.5::numeric(10,0)",-2m) -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13747378/npgsql/1706614870129/1706615156191)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #118414 roachtest: npgsql failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-23.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13762710?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13762710?buildTab=artifacts#/npgsql) on master @ [ed3a25e3c9459cede2f80babbfc9d44a836b6c12](https://github.com/cockroachdb/cockroach/commits/ed3a25e3c9459cede2f80babbfc9d44a836b6c12):


```
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,DefaultTransactionMode).CompositeType -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("1E+4::numeric",10000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V1,BinaryReplicationDataMode,DefaultTransactionMode).CompositeType -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("0.000000000000000000000001::numeric",0.000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ArrayTests(NonMultiplexing).Long_one_dimensional -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ByteaTests(NonMultiplexing).Truncate_array -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1E+24::numeric",1000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.BitStringTests(NonMultiplexing).Array_of_single_bits -  (unexpected)
--- FAIL: Npgsql.Tests.AuthenticationTests(Multiplexing).Password_source_precedence -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ByteaTests(NonMultiplexing).Write_as_Memory -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,StreamingTransactionMode).CompositeType -  (unexpected)
--- FAIL: Npgsql.Tests.Types.MiscTypeTests(NonMultiplexing).Uuid -  (unexpected)
--- FAIL: Npgsql.Tests.ConnectionTests(Multiplexing).Connector_not_initialized_exception -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("1000000000000::numeric",1000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ByteaTests(NonMultiplexing).Write_as_FileStream_long -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("2.4::numeric(10,0)",2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(Multiplexing,Jsonb).As_char_array -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("1E+12::numeric",1000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("0.0000000000000000000000000001::numeric",0.0000000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write_BigInteger("0::numeric",0m) -  (unexpected)
--- FAIL: Npgsql.Tests.SecurityTests.DataSource_UserCertificateValidationCallback_is_invoked(True) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(Multiplexing,Json).As_string_with_GetTextReader -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1E+24::numeric",1000000000000000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ByteaTests(NonMultiplexing).Write_as_MemoryStream_truncated -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write_BigInteger("2::numeric(10,2)",2.00m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("-79228162514264337593543950335::numeric",decimal.MinValue) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read_BigInteger("-1.4::numeric(10,0)",-1m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1E-28::numeric",0.0000000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("0.00000000000000000001::numeric",0.00000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.PoolTests.MinPoolSize_equals_MaxPoolSize -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Read("1.2::numeric(10,2)",1.20m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.DateTimeTests.Date_as_DateTime -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(Multiplexing,Jsonb).As_string_long -  (unexpected)
--- PASS: Npgsql.Tests.ReaderOldSchemaTests(Async).IsAutoIncrement - unknown (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V2,DefaultReplicationDataMode,StreamingTransactionMode).Update_for_index_replica_identity -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("11112222000000000000",11112222000000000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.ByteaTests(NonMultiplexing).Write_as_ReadOnlyMemory -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Write("9364907268378377291970000",9364907268378377291970000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.TextTests(Multiplexing).Text_as_array_of_chars -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("1E-24::numeric",0.000000000000000000000001m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("100000000::numeric",100000000m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(Multiplexing).Read("1.2::numeric(10,1)",1.2m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.NumericTests(NonMultiplexing).Write("1111.2222333344445555666677778::numeric",1111.2222333344445555666677778m) -  (unexpected)
--- FAIL: Npgsql.Tests.Types.TextTests(Multiplexing).Text_as_ArraySegment_of_chars -  (unexpected)
--- FAIL: Npgsql.Tests.Types.JsonTests(Multiplexing,Jsonb).As_JsonDocument -  (unexpected)
--- FAIL: Npgsql.Tests.Types.InternalTypeTests(NonMultiplexing).NpgsqlLogSequenceNumber -  (unexpected)
--- FAIL: Npgsql.Tests.Replication.PgOutputReplicationTests(V3,DefaultReplicationDataMode,StreamingTransactionMode).Update_for_index_replica_identity -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-13762710/npgsql/1706715365368/1706715649649)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/conn_executor.go
+++ b/pkg/sql/conn_executor.go
@@ -3950,8 +3950,12 @@ func (ex *connExecutor) handleWaitingForConcurrentSchemaChanges(
 func (ex *connExecutor) initStatementResult(
 	ctx context.Context, res RestrictedCommandResult, ast tree.Statement, cols colinfo.ResultColumns,
 ) error {
-	for _, c := range cols {
-		if err := checkResultType(c.Typ); err != nil {
+	for i, c := range cols {
+		fmtCode, err := res.GetFormatCode(i)
+		if err != nil {
+			return err
+		}
+		if err = checkResultType(c.Typ, fmtCode); err != nil {
 			return err
 		}
 	}

--- a/pkg/sql/conn_io.go
+++ b/pkg/sql/conn_io.go
@@ -837,6 +837,10 @@ type RestrictedCommandResult interface {
 	// ClientComm.createStatementResult.
 	ResetStmtType(stmt tree.Statement)
 
+	// GetFormatCode returns the format code that will be used to serialize the
+	// data in the provided column when sending messages to the client.
+	GetFormatCode(colIdx int) (pgwirebase.FormatCode, error)
+
 	// AddRow accumulates a result row.
 	//
 	// The implementation cannot hold on to the row slice; it needs to make a
@@ -1103,6 +1107,13 @@ func (r *streamingCommandResult) SendNotice(ctx context.Context, notice pgnotice
 // ResetStmtType is part of the RestrictedCommandResult interface.
 func (r *streamingCommandResult) ResetStmtType(stmt tree.Statement) {
 	panic("unimplemented")
+}
+
+// GetFormatCode is part of the sql.RestrictedCommandResult interface.
+func (r *streamingCommandResult) GetFormatCode(colIdx int) (pgwirebase.FormatCode, error) {
+	// Rows aren't serialized in the streamingCommandResult, so this format code
+	// doesn't really matter - return the default.
+	return pgwirebase.FormatText, nil
 }
 
 // AddRow is part of the RestrictedCommandResult interface.

--- a/pkg/sql/exec_util.go
+++ b/pkg/sql/exec_util.go
@@ -83,6 +83,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgnotice"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgwirebase"
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgwirecancel"
 	"github.com/cockroachdb/cockroach/pkg/sql/physicalplan"
 	"github.com/cockroachdb/cockroach/pkg/sql/querycache"
@@ -109,6 +110,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/upgrade/upgradebase"
 	"github.com/cockroachdb/cockroach/pkg/util/bitarray"
 	"github.com/cockroachdb/cockroach/pkg/util/duration"
+	"github.com/cockroachdb/cockroach/pkg/util/errorutil/unimplemented"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/metric"
@@ -1978,7 +1980,7 @@ func golangFillQueryArguments(args ...interface{}) (tree.Datums, error) {
 
 // checkResultType verifies that a table result can be returned to the
 // client.
-func checkResultType(typ *types.T) error {
+func checkResultType(typ *types.T, fmtCode pgwirebase.FormatCode) error {
 	// Compare all types that can rely on == equality.
 	switch typ.Family() {
 	case types.UnknownFamily:
@@ -2011,6 +2013,11 @@ func checkResultType(typ *types.T) error {
 	case types.EnumFamily:
 	case types.VoidFamily:
 	case types.ArrayFamily:
+		if fmtCode == pgwirebase.FormatBinary && typ.ArrayContents().Family() == types.ArrayFamily {
+			return unimplemented.NewWithIssueDetail(32552,
+				"result", "unsupported binary serialization of multidimensional arrays",
+			)
+		}
 		// Note that we support multidimensional arrays in some cases (e.g.
 		// array_agg with arrays as inputs) but not in others (e.g. CREATE).
 		// Here we allow them in all cases and rely on each unsupported place to

--- a/pkg/sql/pgwire/command_result.go
+++ b/pkg/sql/pgwire/command_result.go
@@ -216,6 +216,29 @@ func (r *commandResult) SetError(err error) {
 	r.err = err
 }
 
+// GetFormatCode is part of the sql.RestrictedCommandResult interface.
+func (r *commandResult) GetFormatCode(colIdx int) (pgwirebase.FormatCode, error) {
+	fmtCode := pgwirebase.FormatText
+	if r.formatCodes != nil {
+		if colIdx >= len(r.formatCodes) {
+			if len(r.formatCodes) == 1 && r.cmdCompleteTag == "EXPLAIN" {
+				// For EXPLAIN ANALYZE statements the format codes describe how
+				// to serialize the EXPLAIN ANALYZE output (which has just one
+				// column) and not the result of the query being executed (which
+				// can have any number of columns). In such a scenario we won't
+				// serialize the data rows (since we will produce stringified
+				// EXPLAIN ANALYZE output and the rows will be actually
+				// discarded by the DistSQLReceiver), so the format code here
+				// doesn't matter.
+				return fmtCode, nil
+			}
+			return 0, errors.AssertionFailedf("could not find format code for column %d in %v", colIdx, r.formatCodes)
+		}
+		fmtCode = r.formatCodes[colIdx]
+	}
+	return fmtCode, nil
+}
+
 // beforeAdd should be called before rows are buffered.
 func (r *commandResult) beforeAdd() error {
 	r.assertNotReleased()

--- a/pkg/sql/pgwire/conn.go
+++ b/pkg/sql/pgwire/conn.go
@@ -881,12 +881,9 @@ func (c *conn) bufferRow(ctx context.Context, row tree.Datums, r *commandResult)
 	c.msgBuilder.initMsg(pgwirebase.ServerMsgDataRow)
 	c.msgBuilder.putInt16(int16(len(row)))
 	for i, col := range row {
-		fmtCode := pgwirebase.FormatText
-		if r.formatCodes != nil {
-			if i >= len(r.formatCodes) {
-				return errors.AssertionFailedf("could not find format code for column %d in %v", i, r.formatCodes)
-			}
-			fmtCode = r.formatCodes[i]
+		fmtCode, err := r.GetFormatCode(i)
+		if err != nil {
+			return err
 		}
 		switch fmtCode {
 		case pgwirebase.FormatText:
@@ -926,12 +923,9 @@ func (c *conn) bufferBatch(ctx context.Context, batch coldata.Batch, r *commandR
 			c.msgBuilder.initMsg(pgwirebase.ServerMsgDataRow)
 			c.msgBuilder.putInt16(width)
 			for vecIdx := 0; vecIdx < len(c.vecsScratch.Vecs); vecIdx++ {
-				fmtCode := pgwirebase.FormatText
-				if r.formatCodes != nil {
-					if vecIdx >= len(r.formatCodes) {
-						return errors.AssertionFailedf("could not find format code for column %d in %v", vecIdx, r.formatCodes)
-					}
-					fmtCode = r.formatCodes[vecIdx]
+				fmtCode, err := r.GetFormatCode(vecIdx)
+				if err != nil {
+					return err
 				}
 				switch fmtCode {
 				case pgwirebase.FormatText:

--- a/pkg/sql/pgwire/conn.go
+++ b/pkg/sql/pgwire/conn.go
@@ -898,10 +898,10 @@ func (c *conn) bufferRow(ctx context.Context, row tree.Datums, r *commandResult)
 		}
 	}
 	if err := c.msgBuilder.finishMsg(&c.writerState.buf); err != nil {
-		return errors.NewAssertionErrorWithWrappedErrf(err, "unexpected err from buffer")
+		return err
 	}
 	if err := c.maybeFlush(r.pos, r.bufferingDisabled); err != nil {
-		return errors.NewAssertionErrorWithWrappedErrf(err, "unexpected err from buffer")
+		return err
 	}
 	c.maybeReallocate()
 	return nil
@@ -943,7 +943,7 @@ func (c *conn) bufferBatch(ctx context.Context, batch coldata.Batch, r *commandR
 				}
 			}
 			if err := c.msgBuilder.finishMsg(&c.writerState.buf); err != nil {
-				panic(fmt.Sprintf("unexpected err from buffer: %s", err))
+				return err
 			}
 			if err := c.maybeFlush(r.pos, r.bufferingDisabled); err != nil {
 				return err

--- a/pkg/sql/pgwire/testdata/pgtest/array
+++ b/pkg/sql/pgwire/testdata/pgtest/array
@@ -30,3 +30,31 @@ ReadyForQuery
 {"Type":"DataRow","Values":[{"text":"{key1,subkey1}"},{"text":"{11,22}"},{"text":"{UnQuoted,Quoted}"}]}
 {"Type":"CommandComplete","CommandTag":"SELECT 1"}
 {"Type":"ReadyForQuery","TxStatus":"I"}
+
+# Regression test for panicking due to not having binary serialization of
+# multidimensional arrays (#118206).
+# "ResultFormatCodes": [1] = binary
+send
+Parse {"Name": "s", "Query": "SELECT ARRAY[ARRAY[1], ARRAY[2]]"}
+Bind {"PreparedStatement": "s", "ResultFormatCodes": [1]}
+Execute
+Sync
+----
+
+until crdb_only
+ErrorResponse
+ReadyForQuery
+----
+{"Type":"ParseComplete"}
+{"Type":"BindComplete"}
+{"Type":"ErrorResponse","Code":"0A000"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+
+until noncrdb_only
+ReadyForQuery
+----
+{"Type":"ParseComplete"}
+{"Type":"BindComplete"}
+{"Type":"DataRow","Values":[{"binary":"0000000200000000000000170000000200000001000000010000000100000004000000010000000400000002"}]}
+{"Type":"CommandComplete","CommandTag":"SELECT 1"}
+{"Type":"ReadyForQuery","TxStatus":"I"}

--- a/pkg/sql/pgwire/testdata/pgtest/prepare
+++ b/pkg/sql/pgwire/testdata/pgtest/prepare
@@ -105,10 +105,42 @@ ReadyForQuery
 {"Type":"CommandComplete","CommandTag":"EXPLAIN"}
 {"Type":"ReadyForQuery","TxStatus":"I"}
 
+# "ResultFormatCodes": [1] = binary
+send
+Parse {"Name": "e2", "Query": "EXPLAIN SELECT i, i + 1, i + 2 FROM generate_series(1,10) AS g(i);"}
+Bind {"DestinationPortal": "pe2", "PreparedStatement": "e2", "ResultFormatCodes": [1]}
+Execute {"Portal": "pe2"}
+Sync
+----
+
+until ignore=DataRow
+ReadyForQuery
+----
+{"Type":"ParseComplete"}
+{"Type":"BindComplete"}
+{"Type":"CommandComplete","CommandTag":"EXPLAIN"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+
 send
 Parse {"Name": "ea", "Query": "EXPLAIN ANALYZE SELECT generate_series(1,10);"}
 Bind {"DestinationPortal": "pea", "PreparedStatement": "ea"}
 Execute {"Portal": "pea"}
+Sync
+----
+
+until ignore=DataRow
+ReadyForQuery
+----
+{"Type":"ParseComplete"}
+{"Type":"BindComplete"}
+{"Type":"CommandComplete","CommandTag":"EXPLAIN"}
+{"Type":"ReadyForQuery","TxStatus":"I"}
+
+# "ResultFormatCodes": [1] = binary
+send
+Parse {"Name": "ea2", "Query": "EXPLAIN ANALYZE SELECT i, i + 1, i + 2 FROM generate_series(1,10) AS g(i);"}
+Bind {"DestinationPortal": "pea2", "PreparedStatement": "ea2", "ResultFormatCodes": [1]}
+Execute {"Portal": "pea2"}
 Sync
 ----
 

--- a/pkg/sql/pgwire/write_buffer.go
+++ b/pkg/sql/pgwire/write_buffer.go
@@ -143,7 +143,9 @@ func (b *writeBuffer) putInt64(v int64) {
 }
 
 func (b *writeBuffer) putInt32AtIndex(index int, v int32) {
-	binary.BigEndian.PutUint32(b.wrapped.Bytes()[index:index+4], uint32(v))
+	if b.err == nil {
+		binary.BigEndian.PutUint32(b.wrapped.Bytes()[index:index+4], uint32(v))
+	}
 }
 
 func (b *writeBuffer) putErrFieldMsg(field pgwirebase.ServerErrFieldType) {

--- a/pkg/sql/plan_opt.go
+++ b/pkg/sql/plan_opt.go
@@ -31,6 +31,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/parser/statements"
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgwirebase"
 	"github.com/cockroachdb/cockroach/pkg/sql/physicalplan"
 	"github.com/cockroachdb/cockroach/pkg/sql/querycache"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/eval"
@@ -178,7 +179,11 @@ func (p *planner) prepareUsingOptimizer(
 		colMeta := md.ColumnMeta(col.ID)
 		resultCols[i].Name = col.Alias
 		resultCols[i].Typ = colMeta.Type
-		if err := checkResultType(resultCols[i].Typ); err != nil {
+		// At PREPARE time we don't know yet which format the client will
+		// request (this is only known at BIND time), so we optimistically
+		// assume that it'll be TEXT (which is the default).
+		fmtCode := pgwirebase.FormatText
+		if err = checkResultType(resultCols[i].Typ, fmtCode); err != nil {
 			return 0, err
 		}
 		// If the column came from a table, set up the relevant metadata.

